javascripts/discourse/components/table-header-renderer.js

```diff
@@ -0,0 +1,16 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+
+export default class TableHeaderRenderer extends Component {
+  @tracked filteredUserFields = [];
+
+  constructor() {
+    super(...arguments);
+    const idsToShow = settings.user_custom_field_ids_to_show
+      .split("|")
+      .map((id) => parseInt(id, 10));
+    this.filteredUserFields = this.args.userFields.filter((field) =>
+      idsToShow.includes(field.id)
+    );
+  }
+}
```

Comment:
You don't update `filteredUserFields` after initialization so it doesn't have to be tracked (and initialization can be moved out of the constructor. For example:

```suggestion
import Component from "@glimmer/component";

export default class TableHeaderRenderer extends Component {
  idsToShow = settings.user_custom_field_ids_to_show
    .split("|")
    .map((id) => parseInt(id, 10));
  };
  filteredUserFields = this.args.userFields.filter((field) =>
    this.idsToShow.includes(field.id)
  );
}
```

Or you could move `filteredUserFields` to a getter. I'll add an example in a comment for another file.